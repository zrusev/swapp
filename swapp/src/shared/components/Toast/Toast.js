import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

const kf = keyframes`
    0%{
        opacity:0;
        transform: translateY(20em);
    }

    100%{
        opacity:1;
        transform: translateY(0);
    }
`;

const Container = styled.div`
  position: fixed;
  bottom: 2vh;
  left: 1vw;
  border-radius: 4px;
  box-shadow: #310808 2px 2px 7px;
  background-color: ${({ theme: { background } }) => background};
  padding: 0.5em;
  color: ${({ theme: { primaryHeadingFontColor } }) => primaryHeadingFontColor};

  opacity: 1;
  animation: ${kf} 500ms cubic-bezier(0.23, 0.82, 0.16, 1.46);
  animation-iteration-count: 1;
`;

const Toast = ({ children }) => (
  <Container>
    <p>{children}</p>
  </Container>
);

Toast.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};

export default Toast;
