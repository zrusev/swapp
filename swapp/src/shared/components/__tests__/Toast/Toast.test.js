import React from 'react';
import { shallow } from 'enzyme';

import Toast from '../../Toast/Toast';

describe('should Toast component be styled', () => {
  const wrapper = shallow(<Toast>message</Toast>);

  it('should be mount', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should load children', () => {
    expect(wrapper.text()).toEqual('message');
  });

  it('should load children html', () => {
    expect(wrapper.containsMatchingElement('message')).toBeTruthy();
  });
});
