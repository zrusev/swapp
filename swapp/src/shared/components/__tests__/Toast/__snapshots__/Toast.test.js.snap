// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Toast component be styled should be mount 1`] = `
<styled.div>
  <p>
    message
  </p>
</styled.div>
`;
